### ISSUE TITLE
F5 onboarding is failing

### ISSUE THREAD

**Issue by KodiakF5Team** [F5Networks/f5-bigip-runtime-init#49]
We are trying to do onboarding with IPv6 , As per azure IPv6 can be defined as secondary IP,
it is failing with error  "Cannot read property '0' of undefined"

[error log.txt](https://github.com/F5Networks/f5-bigip-runtime-init/files/10110892/error.log.txt)
[runtime_init.conf.txt](https://github.com/F5Networks/f5-bigip-runtime-init/files/10110893/runtime_init.conf.txt)
[bigIpRuntimeInit.log](https://github.com/F5Networks/f5-bigip-runtime-init/files/10110895/bigIpRuntimeInit.log)

**Comment by mikeshimkus** [comment]
Hi @KodiakF5Team, yes, I can see in the code where that wouldn't work.  I have created internal issue #3235 to track the fix. Can you try the following as a workaround:

```
- name: SELF_IP_EXTERNAL7
  type: url
  query: ipv6.ipAddress[0]
  value: 'http://169.254.169.254/metadata/instance/network/interface/6?api-version=2017-08-01'
  headers:
    - name: Metadata
      value: true
```

**Comment by KodiakF5Team** [comment]
Hi Mike

Thanks for sharing update!
We have tried the workaround and the onboarding is failing again with below error:
[bigIpRuntimeInit.log](https://github.com/F5Networks/f5-bigip-runtime-init/files/10132991/bigIpRuntimeInit.log)
[error log](https://github.com/F5Networks/f5-bigip-runtime-init/files/10132993/error.log)


    "post_onboard_enabled": {
      "commands": 0,
      "postHooks": 0
    },
    "result": "FAILURE",
    "resultSummary": "{\"code\":400,\"message\":\"request failed with null exception\",\"referer\":\"Unknown\",\"restOperationId\":1245284,\"kind\":\":res
[runtime-init.txt](https://github.com/F5Networks/f5-bigip-runtime-init/files/10133015/runtime-init.txt)
[runtime-init.txt](https://github.com/F5Networks/f5-bigip-runtime-init/files/10133025/runtime-init.txt)
terrorresponse\"}",
    "startTime": "2022-12-01T13:26:52.139Z",
    "endTime": "2022-12-01T13:29:31.417Z",
    "installParams": [
      {
        "key": "templateName",
        "value": "v2.2.0.0/examples/modules/bigip-standalone/bigip.json"
      }

I have attached the log file for reference.

**Comment by KodiakF5Team** [comment]
@shyawnkarim @mikeshimkus Hi,

Can you please help me to resolve this IPv6 support, if any suggestion let me know?

**Comment by mikeshimkus** [comment]
Hi @KodiakF5Team, this will require a code change which we are tracking with issue #3235. We'll let you know when the fix has been released.

**Comment by G-gonzalezjimenez** [comment]
Hi @KodiakF5Team  could you contact us through email internally so we can talk about this issue. Thank you

**Comment by KodiakF5Team** [comment]
Hi Guillermo Gonzalez,

Sure , let me know if any other info is required.

*Thanks and Regards*

*Jhaunu Gupta*
***@***.***

*Kodiak*, a Motorola Solutions Company

+91 9945024062  |  kodiakptt.com  |  motorolasolutions.com

[image: Kodiak, a Motorola Solutions Company] <https://kodiakptt.com/>


On Wed, Dec 7, 2022 at 1:07 AM 'Guillermo Gonzalez' via IDC F5 Team <
***@***.***> wrote:

> Hi @KodiakF5Team
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_KodiakF5Team&d=DwMCaQ&c=q3cDpHe1hF8lXU5EFjNM_C93KOmcBXCBnhee2v6PYlc&r=4qYRzE1rADJxfns4kXa16tNOljbgqof1ftGheCb-3m4&m=6i2BVHVyaCHASHXZSOtx9ckhZLmWiDOPcPfaMdQ-nGlayZI2BZJRXd1s17bi0Pqh&s=heZVzAya_oXtxMpAtLHDXHZOcUiJWgQNiuuTsEIFY8g&e=>
> could you contact us through email internally so we can talk about this
> issue. Thank you
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_F5Networks_f5-2Dbigip-2Druntime-2Dinit_issues_49-23issuecomment-2D1339910583&d=DwMCaQ&c=q3cDpHe1hF8lXU5EFjNM_C93KOmcBXCBnhee2v6PYlc&r=4qYRzE1rADJxfns4kXa16tNOljbgqof1ftGheCb-3m4&m=6i2BVHVyaCHASHXZSOtx9ckhZLmWiDOPcPfaMdQ-nGlayZI2BZJRXd1s17bi0Pqh&s=UlMcRCjlDpuAfJ5jYzgixigz9pITG0byIKpspRdfxBA&e=>,
> or unsubscribe
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AUVQED3N4GONHRJUIK26HQ3WL6IZDANCNFSM6AAAAAASOFLVS4&d=DwMCaQ&c=q3cDpHe1hF8lXU5EFjNM_C93KOmcBXCBnhee2v6PYlc&r=4qYRzE1rADJxfns4kXa16tNOljbgqof1ftGheCb-3m4&m=6i2BVHVyaCHASHXZSOtx9ckhZLmWiDOPcPfaMdQ-nGlayZI2BZJRXd1s17bi0Pqh&s=AbrwIIxLm39UNizZFhYLM3-o9zNKR5AdMpWBC3vY7aE&e=>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 


*For more information on how and why we collect your personal 
information, please visit our Privacy Policy 
<https://www.motorolasolutions.com/en_us/about/privacy-policy.html?elqTrackId=8980d888905940e39a2613a7a3dcb0a7&elqaid=2786&elqat=2#privacystatement>.*

**Comment by KodiakF5Team** [comment]
HI Guillermo 

Can you please let us know any method or dl to contact you for the issue, or do we need to open a F5 Support case for it?

Regards
Aayush Narayan

**Comment by shyawnkarim** [comment]
Closing. 

This bug was resolved with [Release 1.5.2](https://github.com/F5Networks/f5-bigip-runtime-init/releases/tag/1.5.2).
